spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -3497,6 +3497,280 @@ void stopImmediately() throws InterruptedException {
 		verify(consumer).commitSync(eq(Map.of(new TopicPartition("foo", 0), new OffsetAndMetadata(1L))), any());
 	}
 
+	@Test
+	public void shouldInvokeRecordInterceptorSuccess() throws Exception {
+		ConsumerFactory<Integer, String> cf = mock(ConsumerFactory.class);
+		Consumer<Integer, String> consumer = mock(Consumer.class);
+		given(cf.createConsumer(eq("grp"), eq("clientId"), isNull(), any())).willReturn(consumer);
+		ConsumerRecord<Integer, String> firstRecord = new ConsumerRecord<>("foo", 0, 0L, 1, "foo");
+		ConsumerRecord<Integer, String> secondRecord = new ConsumerRecord<>("foo", 0, 1L, 1, "bar");
+		Map<TopicPartition, List<ConsumerRecord<Integer, String>>> records = new HashMap<>();
+		records.put(new TopicPartition("foo", 0), List.of(firstRecord, secondRecord));
+		ConsumerRecords<Integer, String> consumerRecords = new ConsumerRecords<>(records);
+		given(consumer.poll(any(Duration.class))).willAnswer(i -> {
+			Thread.sleep(50);
```

Comment:
Do we really need this sleep?
Not clear what further logic expects some delays with records delivery...

Reply:
Yes; otherwise the consumer thread will call it in a tight loop.

Reply:
OK. won't another `willAnswer(i -> null)` after this one help?
According Mockito docs only the last stub is used for the rest of call.
We probably need to have a idle on the container in addition...
But yeah, we can live it as is: just wanted to share with you a logic possible via `.willAnswer(...).willAnswer(...)` chain.

Reply:
No; returning null will not prevent the tight loop; we could add a latch so the second poll blocks until the test completes but I don't feel it's that important - there are many tests with this logic; we can put in on the slate for improvement later.